modules/P19_bsd_ufs_mounter.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+# EMBA - EMBEDDED LINUX ANALYZER
+#
+# Copyright 2020-2022 Siemens Energy AG
+# Copyright 2020-2022 Siemens AG
+#
+# EMBA comes with ABSOLUTELY NO WARRANTY. This is free software, and you are
+# welcome to redistribute it under the terms of the GNU General Public License.
+# See LICENSE file for usage of this software.
+#
+# EMBA is licensed under GPLv3
+#
+# Author(s): Michael Messner, Pascal Eckmann
+
+# Description: Mounts and extracts BSD UFS images
+# Pre-checker threading mode - if set to 1, these modules will run in threaded mode
+export PRE_THREAD_ENA=0
+
+P19_bsd_ufs_mounter() {
+  module_log_init "${FUNCNAME[0]}"
+  NEG_LOG=0
+  if [[ "$BSD_UFS" -eq 1 ]]; then
+    module_title "BSD UFS filesystem extractor"
+    pre_module_reporter "${FUNCNAME[0]}"
+
+    print_output "[*] Connect to device $ORANGE$FIRMWARE_PATH$NC"
+
+    EXTRACTION_DIR="$LOG_DIR"/firmware/ufs_mount_filesystem/
+
+    ufs_extractor "$FIRMWARE_PATH" "$EXTRACTION_DIR"
+
+    export FIRMWARE_PATH="$LOG_DIR"/firmware/
+    NEG_LOG=1
+  fi
+  module_end_log "${FUNCNAME[0]}" "$NEG_LOG"
+}
+
+ufs_extractor() {
+  local UFS_PATH_="$1"
+  local EXTRACTION_DIR_="$2"
+  local TMP_UFS_MOUNT="$TMP_DIR""/ufs_mount_$RANDOM"
+  local FILES_UFS_MOUNT
+  local DIRS_UFS_MOUNT
+  sub_module_title "UFS filesystem extractor"
+
+  mkdir -p "$TMP_UFS_MOUNT"
+  print_output "[*] Trying to mount $ORANGE$UFS_PATH_$NC to $ORANGE$TMP_UFS_MOUNT$NC directory"
+  modprobe ufs
+  mount -r -t ufs -o ufstype=ufs2 "$UFS_PATH_" "$TMP_UFS_MOUNT"
+  if mount | grep -q ufs_mount; then
+    print_output "[*] Copying $ORANGE$TMP_UFS_MOUNT$NC to firmware tmp directory ($EXTRACTION_DIR_)"
+    mkdir -p "$EXTRACTION_DIR_"
```

Comment:
Don't you have to add ` > /dev/null || true ` or something like this at the end of each `mkdir` call - I grepped trough the code and there are some more of these missing.